src/cdk-experimental/ui-patterns/listbox/listbox.spec.ts

```diff
@@ -279,21 +280,28 @@ describe('Listbox Pattern', () => {
         expect(listbox.inputs.value()).toEqual(['Apple', 'Apricot']);
       });
 
-      it('should toggle the selected state of the next option on Shift + ArrowDown', () => {
+      it('should select a range of options on Shift + ArrowDown/ArrowUp', () => {
```

Comment:
There may be a test for this already somewhere, but just wanted to point out the case where there's random disabled/readonly items in the middle of your range

Reply:
Finally done after a full 24 hours of tinkering 😂 This sent me down a rabbit hole but I think it was for the better